### PR TITLE
docs: on append, overwrite, delete and z-ordering

### PR DESCRIPTION
**Pull request by MrPowers**
# Description

Adds docs on how to append, overwrite, delete rows, and Z Order Delta tables.

Will add much more detailed pages in the future.  Just getting the high-level skeleton of the docs developed.

### CODE REVIEW

File: docs/usage/appending-overwriting-delta-lake-table.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+|     2 | b        |
+|     3 | c        |
++-------+----------+
+```
+
+Append two additional rows of data to the table:
+
+```python
+df = pd.DataFrame({"num": [8, 9], "letter": ["dd", "ee"]})
+dl.writer.write_deltalake("tmp/some-table", df, mode="append")
```
Comment (ion-elgreco) [inline docs/usage/appending-overwriting-delta-lake-table.md:23]: Should we showcase examples where you import it like as dl. 

I think we generally import directly, right?

File: docs/usage/create-delta-lake-table.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Creating a Delta Lake Table
+
+This section explains how to create a Delta Lake table.
+
+You can easily write a DataFrame to a Delta table.
+
+```python
+import deltalake as dl
```
Comment (ion-elgreco) [inline docs/usage/create-delta-lake-table.md:8]: I think you forgot to remove this import as well

File: docs/usage/create-delta-lake-table.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+from deltalake.writer import write_deltalake
```
Comment (ion-elgreco) [inline docs/usage/create-delta-lake-table.md:9]: We expose write_deltalake also on the deltalake module directly. Also in other docs we import directly from there, maybe do it here as well for consistency

**Review by ion-elgreco — CHANGES_REQUESTED** [review]

File: docs/usage/create-delta-lake-table.md (hunk 3 above)
Comment (MrPowers) [reply docs/usage/create-delta-lake-table.md:9]: I get this error: `cannot import name 'DeltaLake' from 'deltalake'`

You sure `write_deltalake` is exposed on `deltalake`?

File: docs/usage/optimize/delta-lake-z-order.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Delta Lake Z Order
+
+This section explains how to Z Order a Delta table.
+
+Z Ordering colocates similar data in the same files, which allows for more better file skipping and faster queries.
```
Comment (thomasfrederikhoeck) [inline docs/usage/optimize/delta-lake-z-order.md:5]: `more` or `better`, right? not both

Comment (MrPowers) [reply docs/usage/optimize/delta-lake-z-order.md]: Fixed, thanks

File: docs/usage/create-delta-lake-table.md (hunk 3 above)
Comment (MrPowers) [reply docs/usage/create-delta-lake-table.md]: I had a typo. You're right.  Updated.

File: docs/usage/create-delta-lake-table.md (hunk 2 above)
Comment (MrPowers) [reply docs/usage/create-delta-lake-table.md]: Updated.

**Pull request merged**